mypy/meet.py

```diff
@@ -489,17 +489,18 @@ def visit_tuple_type(self, t: TupleType) -> Type:
                 items.append(self.meet(t.items[i], self.s.items[i]))
             # TODO: What if the fallbacks are different?
             return TupleType(items, t.fallback)
-        # meet(Tuple[t1, t2, <...>], Tuple[s, ...]) == Tuple[meet(t1, s), meet(t2, s), <...>].
-        elif (isinstance(self.s, Instance) and
-              (self.s.type.fullname() == 'builtins.tuple' or is_proper_subtype(t, self.s))
-              and self.s.args):
-            return t.copy_modified(items=[meet_types(it, self.s.args[0]) for it in t.items])
+        elif isinstance(self.s, Instance):
+            # meet(Tuple[t1, t2, <...>], Tuple[s, ...]) == Tuple[meet(t1, s), meet(t2, s), <...>].
+            if self.s.type.fullname() == 'builtins.tuple' and self.s.args:
+                return t.copy_modified(items=[meet_types(it, self.s.args[0]) for it in t.items])
+            elif is_proper_subtype(t, self.s):
+                # A named tuple that inherits from a normal class
+                return t
         elif (isinstance(self.s, Instance) and t.fallback.type == self.s.type):
```

Comment:
It looks like the condition can never be true. Either move this to the above elif block or remove?

Reply:
Yes, good catch, I think this is not needed anymore.

Reply:
OK, fixed now!